packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/MultiFrameViewer/MultiFrameViewerContainer.js

```diff
@@ -86,7 +95,15 @@ class MultiFrameViewerContainer extends React.Component {
   }
 
   onFrameChange(frame) {
-    const { setFrame } = this.props
+    const {
+      activeTool,
+      setFrame
+    } = this.props
+
+    if (activeTool?.marks?.size > 0) {
+      activeTool.validate()
```

Comment:
What does `tool.validate()` do? It seems like there might be hidden side effects here.

Reply:
it checks each mark's `isValid` view and deletes the mark if `isValid` returns `false` (i.e. mark is invalid). For the transcription tool that is 
```
get isValid() {
  return self.x1 !== self.x2 || self.y1 !== self.y2
},
```
I had the same feeling about hidden side effects, but I think the underlying issue of invalid marks needing to be deleted (i.e. `tool.validate()` on frame change) applies to all drawing tools, is ok. Alternatively I tested on the workflow with multiple frames and no drawing task/tool and looked ok.

Reply:
Sounds good to me. Deleting marks is the kind of hidden side effect I was thinking of too, but I have no better suggestions how to make it obvious to a casual reader that this code does that.